app/stac_api/views.py

```diff
@@ -344,6 +389,208 @@ def delete(self, request, *args, **kwargs):
+                        if (
+                            isinstance(value, str) and operator in int_operators and
+                            attribute in int_operators
+                        ):
+                            message = f"You are not allowed to compare a string/date ({attribute})"\
+                                      f" with a number operator." \
+                                      f"for string use one of these {str_operators}"
+                            logger.error(message)
+                            raise ValidationError(_(message))
+                        if (
+                            isinstance(value, int) and operator in str_operators and
+                            operator in str_operators
+                        ):
+                            message = f"You are not allowed to compare a number or a date with" \
+                                      f"a string operator." \
+                                      f"For numbers use one of these {int_operators}"
+                            logger.error(message)
+                            raise ValidationError(_(message))
+
+                        # treate date
+                        if attribute in queriable_date_fields:
+                            try:
+                                if isinstance(value, list):
+                                    value = [fromisoformat(i) for i in value]
+                                else:
+                                    value = fromisoformat(value)
+                            except ValueError as error:
+                                message = f"Invalid dateformat: ({error})"
+                                logger.error(message)
+                                raise ValidationError(_(message))
+
+
+                        # __eq does not exist, but = does it as well
+                        if operator == 'eq':
+                            query_filter = f"properties_{attribute}"
+                        else:
+                            query_filter = f"properties_{attribute}__{operator.lower()}"
+
+                        queryset = queryset.filter(**{query_filter: value})
+
+                        logger.debug("query_filter: %s", query_filter)
+                        logger.debug("operator: %s", operator)
+                        logger.debug("value: %s", value)
+                    else:
+                        message = f"Invalid operator in query argument. The operator {operator} " \
+                                  f"is not supported. Use: {operators}"
+                        logger.error(message)
+                        raise ValidationError(_(message))
+            else:
+                message = f"Invalid field in query argument. The field {attribute} is not " \
+                          f"a propertie. Use one of these {queriable_fields}"
+                logger.error(message)
+                raise ValidationError(_(message))
+        return queryset
```

Comment:
FOR NEXT PR: 
- [ ] Separate the parsing logic with the queryset logic. I mean first do the parsing with validation of the json in a method looping over all attributes, then in a second method loop again in the attributes to generate the queryset. Looping twice here is totally acceptable because we should not have big query string with thousand attributes and it would make the code cleaner and easier to understand. 
- [ ] In the parsing method then make sure to not allow duplicate attributes and to only have valid attributes (this would automatically prevent looping over a big structure twice :-))
- [ ] In the parsing method make uses of the multiple errors, instead of stopping at the first errors save the error in a list and at the end of the parsing return `ValidationError(error_list, code='query-invalid')`
- [ ] Due to the preceding we need to avoid to loop over a big numbers of attributes, otherwise a malicious could generate a huge query string for a DoS. So a first check that the `len(json_query) < (number_max_attribute + 5)` and directly stop the validation if this is not true. This way we can make uses of multiples error without the risk of a big loop.
- [ ] Make more explicit debug message, so that we can use them on Kibana (add a context). For example the request could be added as extra.

Reply:
https://github.com/geoadmin/service-stac/pull/181

Reply:
Do not allow duplicate attributes: They will be treated with AND... so I will allow 50 times the same a attribte